web/src/scheduled_messages_popover.js

```diff
@@ -150,11 +152,21 @@ export function initialize() {
         onShow(instance) {
             const formatted_send_later_time =
                 scheduled_messages.get_formatted_selected_send_later_time();
+            // Show an option to keep the current draft and start a new message,
+            // but only when the compose box has a non-empty saved draft for
+            // the current narrow.
+            const current_drafts = drafts.draft_model.get();
+            const narrow_drafts = drafts.filter_drafts_by_compose_box_and_recipient(current_drafts);
+            const show_compose_new_message =
+                narrow_drafts &&
+                compose_state.has_message_content() &&
+                compose_state.has_complete_recipient();
```

Comment:
I don't understand the conditional here -- shouldn't this option be useful regardless of whether you have any drafts matching the view? Like what if you just have some text in the compose box right now (not already saved as a draft) and want to start afresh saving that content as a draft?

Reply:
I'll change this when I make the other change, but I'll wait for your response on that first

Reply:
I changed to this

```js
            // If there's existing text in the composebox, show an option
            // to keep/save the current draft and start a new message.
            const show_compose_new_message = compose_state.has_message_content();
```

Reply:
I actually ended up making a `has_savable_message_content`, which checks if the draft is long enough to save, because I don't want to make false promises of saving a draft when we actually throw away "draft" messages of <= 2 characters.